var/spack/repos/jcsda-emc/packages/gsi-ncdiag/package.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class GsiNcdiag(CMakePackage):
+    """GSI NetCDF Diagnostics Library and Utility Tools"""
+
+    homepage = "https://github.com/NOAA-EMC/GSI-ncdiag"
+    url      = "https://github.com/NOAA-EMC/GSI-ncdiag/archive/refs/tags/v1.0.0.tar.gz"
+
+    maintainers = ['kgerheiser']
+
+    version('1.0.0', sha256='7251d6139c2bc1580db5f7f019e10a4c73d188ddd52ccf21ecc9e39d50a6af51')
+
+    variant('serial', default=True, description='Enable Serial NetCDF diagnostics')
+
+    depends_on('mpi')
+    depends_on('netcdf-fortran@4.5.2:')
+
+    def cmake_args(self):
+        args = [
+          self.define_from_variant('ENABLE_NCDIAG_SERIAL', 'serial')
+        ]
+
+        args.append(self.define('CMAKE_C_COMPILER', self.spec['mpi'].mpicc))
```

Comment:
Is this needed for a Fortran package?
`cmake` will ignore it.

Reply:
Maybe not, but when using `find_package(MPI)` it's necessary to explicitly set MPI as the compiler because CMake will not pick up Intel MPI from Spack (the prefix Spack sets is not the directory of `bin`/etc). And MPI::MPI_Fortran requires C MPI.

Reply:
The usual reply from the spack developers is that none of this should be needed and that it may result in overlinking. But we did have to add it to a few packages in the past.